android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -336,6 +337,13 @@ constructor(
     fun formatDate(date: Date, expectedFormat: String = "E, MMM dd yyyy"): String {
       return date.formatDate(expectedFormat)
     }
+
+    fun getReadyTask(list: List<Task>): Task? =
+      if (list.isNotEmpty()) {
+        list.find { it.status == Task.TaskStatus.READY }
+      } else {
+        null
+      }
   }
```

Comment:
This function is not generic. I think we can address this use case with JEXL expression. It's possible to loop through a list using JEXL

Reply:
[JEXL Syntax](https://commons.apache.org/proper/commons-jexl/reference/syntax.html)